src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -87,12 +93,61 @@ public static IHostBuilder AddScriptHost(this IHostBuilder builder, ScriptApplic
+            builder.AddScriptHostCore(applicationOptions, configureWebJobs, loggerFactory);
+
+            // Allow FunctionsStartup to add configuration after all other configuration is registered.
+            builder.ConfigureAppConfiguration((context, configBuilder) =>
+            {
+                // Pre-build configuration here to load bundles and to store for later validation.
+                var config = configBuilder.Build();
+                var extensionBundleOptions = GetExtensionBundleOptions(config);
+                var bundleManager = new ExtensionBundleManager(extensionBundleOptions, SystemEnvironment.Instance, loggerFactory);
+                var metadataServiceProvider = applicationOptions.RootServiceProvider.GetService<IFunctionMetadataProvider>();
+                var locator = new ScriptStartupTypeLocator(applicationOptions.ScriptPath, loggerFactory.CreateLogger<ScriptStartupTypeLocator>(), bundleManager, metadataServiceProvider, metricsLogger);
+
+                // The locator (and thus the bundle manager) need to be created now in order to configure app configuration.
+                // Store them so they do not need to be re-created later when configuring services.
+                context.Properties[BundleManagerKey] = bundleManager;
+                context.Properties[StartupTypeLocatorKey] = locator;
+
+                if (context.Properties[DelayedConfigurationActionKey] is Action<IWebJobsConfigurationStartupTypeLocator> delayedConfigAction)
+                {
+                    context.Properties.Remove(DelayedConfigurationActionKey);
+
+                    // store the snapshot for validation later
+                    context.Properties[ConfigurationSnapshotKey] = config;
+
+                    delayedConfigAction(locator);
+                }
+            });
+
+            return builder;
         }
 
-        public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptApplicationHostOptions applicationHostOptions, IMetricsLogger metricsLogger, Action<IWebJobsBuilder> configureWebJobs = null, ILoggerFactory loggerFactory = null)
+        public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptApplicationHostOptions applicationHostOptions, Action<IWebJobsBuilder> configureWebJobs = null, ILoggerFactory loggerFactory = null)
         {
             var skipHostInitialization = builder.Properties.ContainsKey(ScriptConstants.SkipHostInitializationKey);
+
+            builder.ConfigureServices((context, services) =>
+            {
+                services.AddSingleton<IHostedService>(s =>
+                {
+                    var environment = s.GetService<IEnvironment>();
+                    var originalConfig = context.Properties.GetAndRemove<IConfigurationRoot>(ConfigurationSnapshotKey);
+
+                    if (environment.IsWindowsConsumption() || environment.IsCoreTools())
```

Comment:
We can probably add a check for Linux Consumption as well, and maybe even Elastic premium skus in both environment.

Reply:
Yes need to add ElasticPremium here too -- good catch.